src/optimizer/plugins/SimplifyArrayIndexingOptimizer.ts

```diff
@@ -0,0 +1,170 @@
+/**
+ * Populates each PathPart in an array of PathParts with a prefix containing the assembled path up to that element
+ * @param { path: PathPart[] } parsedPath - An array of PathParts
+ */
+function addPrefixes(parsedPath: fhirtypes.PathPart[]) {
+  parsedPath.forEach((element: fhirtypes.PathPart, elementIndex: number) => {
+    // Add a prefix to the current element containing previously parsed rule elements
+    element.prefix = utils.assembleFSHPath(parsedPath.slice(0, elementIndex));
+  });
+}
+
+/**
+ * Converts numeric indeces on a PathPart into soft indexing characters
+ * @param {PathPart} element - A single element in a rules path
+ * @param {Map<string, number} pathMap - A map containing an element's name as the key and that element's updated index as the value
+ * @param {Map<string, PathPart[]} singletonArrayElements - A map containing a string unique to each element of type array as the key, and an array of PathParts as the value
+ */
+function applySoftIndexing(element: fhirtypes.PathPart, pathMap: Map<string, number>) {
+  // Must account for a pathPart's base name, prior portions of the path, as well as any slices it's contained in.
+  const mapName = `${element.prefix ?? ''}.${element.base}|${(element.slices ?? []).join('|')}`;
+  const indexRegex = /^[0-9]+$/;
+
+  const currentNumericBracket = element.brackets?.find(bracket => indexRegex.test(bracket));
+  if (!currentNumericBracket) return; // If the element is not an array, exit the function
+
+  const bracketIndex = element.brackets.indexOf(currentNumericBracket);
+
+  if (!pathMap.has(mapName)) {
+    pathMap.set(mapName, parseInt(currentNumericBracket));
+  } else {
+    const previousNumericIndex = pathMap.get(mapName);
+    if (parseInt(currentNumericBracket) === previousNumericIndex + 1) {
+      element.brackets[bracketIndex] = '+';
+      pathMap.set(mapName, previousNumericIndex + 1);
+    } else if (parseInt(currentNumericBracket) === previousNumericIndex) {
+      element.brackets[bracketIndex] = '=';
+    } else {
+      pathMap.set(mapName, parseInt(currentNumericBracket));
+    }
+  }
+}
+
+// Removes '[0]' and '[=]' indexing from elements with a single value in their array
+function removeZeroIndices(parsedPaths: any[]) {
+  const regularPathElements = _.flatten(parsedPaths.map(rule => rule.path));
+  const caretPathElements = _.flatten(parsedPaths.map(rule => rule.caretPath));
+
+  const referencePathElements = _.cloneDeep(regularPathElements);
+  const referenceCaretPathElements = _.cloneDeep(caretPathElements);
+
+  regularPathElements.forEach((element: fhirtypes.PathPart) => {
+    if (!element.brackets) {
+      return;
+    }
+    const filteredElements = referencePathElements.filter(
+      e =>
+        e.base === element.base &&
+        e.prefix === element.prefix &&
+        e.brackets &&
+        !(e.brackets.includes('0') || e.brackets.includes('='))
+    );
+    if (filteredElements.length === 0) {
+      if (element.brackets.includes('0')) {
```

Comment:
Something is still going wrong with CapabilityStatements on US Core. I'm not entirely sure, but I think it is likely due to this line, since the error starts at index `10`, and this line is only looking for `0` to be included, which is included in `10`.

Reply:
I've detailed what was going on here in a separate comment, this should be addressed in `ba71e9f`.